cot/src/form/fields.rs

```diff
@@ -199,6 +200,81 @@ impl AsFormField for PasswordHash {
     }
 }
 
+impl_form_field!(EmailField, EmailFieldOptions, "an email");
+
+/// Custom options for [`EmailField`]
+#[derive(Debug, Default, Copy, Clone)]
+pub struct EmailFieldOptions {
+    /// The maximum length of the field used to set the `maxlength` attribute
+    /// in the HTML input element.
+    pub max_length: Option<u32>,
+    /// The minimum length of the field used to set the `minlength` attribute
+    /// in the HTML input element.
+    pub min_length: Option<u32>,
+}
+
+impl Display for EmailField {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        let mut tag = HtmlTag::input("email");
+        tag.attr("name", self.name());
+        tag.attr("id", self.id());
+        if self.options.required {
+            tag.bool_attr("required");
+        }
+        if let Some(max_length) = self.custom_options.max_length {
+            tag.attr("maxlength", &max_length.to_string());
+        }
+        if let Some(min_length) = self.custom_options.min_length {
+            tag.attr("minlength", &min_length.to_string());
+        }
+        if let Some(value) = &self.value {
+            tag.attr("value", value);
+        }
+
+        write!(f, "{}", tag.render())
+    }
+}
+
+impl AsFormField for Email {
+    type Type = EmailField;
+
+    fn clean_value(field: &Self::Type) -> Result<Self, FormFieldValidationError>
+    where
+        Self: Sized,
+    {
+        let value = check_required(field)?;
+        let opts = &field.custom_options;
+
+        if let (Some(min), Some(max)) = (opts.min_length, opts.max_length) {
+            if min > max {
+                return Err(FormFieldValidationError::from_string(format!(
+                    "min_length ({min}) exceeds max_length ({max})"
+                )));
+            }
```

Comment:
Ah, it would probably make sense to have some sort of mechanism to enforce this when constructing the field, rather than when cleaning the value. This is obviously out of scope of this PR, but I'll create an issue to keep this in the roadmap.